packages/carbon-web-components/src/components/text-input/text-input-story.ts

```diff
@@ -0,0 +1,177 @@
+        <cds-layer>
+          <cds-text-input>
+            <span slot="label-text">Third layer</span>
+            <span slot="helper-text">Optional help text</span>
+          </cds-text-input>
+        </cds-layer>
+      </cds-layer>
+    </cds-layer>
+  `;
+};
+
+export const Playground = (args) => {
+  const {
+    disabled,
+    enableCounter,
+    helperText,
+    hideLabel,
+    inline,
+    invalid,
+    invalidText,
+    labelText,
+    maxCount,
+    placeholder,
+    playgroundWidth,
+    readonly,
+    size,
+    type,
+    value,
+    warn,
+    warnText,
+    onInput,
+  } = args?.[`${prefix}-input`] ?? {};
+  return html`
+    <div style="width: ${playgroundWidth}px;">
+      <cds-text-input
+        ?disabled="${disabled}"
+        ?enable-counter="${ifDefined(enableCounter)}"
+        helper-text="${ifDefined(helperText)}"
+        ?hide-label="${hideLabel}"
+        ?inline="${inline}"
+        ?invalid="${invalid}"
+        invalid-text="${ifDefined(invalidText)}"
+        label="${ifDefined(labelText)}"
+        max-count="${ifDefined(maxCount)}"
+        placeholder="${ifDefined(placeholder)}"
+        ?readonly="${ifDefined(readonly)}"
+        size="${ifDefined(size)}"
+        type="${ifDefined(type)}"
+        value="${ifDefined(value)}"
+        ?warn="${ifDefined(warn)}"
+        warn-text="${ifDefined(warnText)}"
+        @input="${onInput}">
+      </cds-text-input>
+    </div>
+  `;
+};
+
+Playground.parameters = {
+  knobs: {
+    [`${prefix}-input`]: () => ({
+      disabled: boolean('Disabled (disabled)', false),
+      enableCounter: boolean('Enable counter (enable-counter)', false),
+      helperText: textNullable('Helper text (helper-text)', 'Helper text'),
+      hideLabel: boolean('Hide label (hide-label)', false),
+      inline: boolean('Inline (inline)', false),
+      invalid: boolean('Invalid (invalid)', false),
+      invalidText: textNullable(
+        'Invalid text (invalid-text)',
+        'Error message goes here'
+      ),
+      labelText: textNullable('Label text (label)', 'Label text'),
+      maxCount: textNullable('Max count (max-count)', '100'),
+      placeholder: textNullable(
+        'Placeholder (placeholder)',
+        'Placeholder text'
+      ),
+      playgroundWidth: number('Playground width', 300, {
+        range: true,
+        min: 300,
+        max: 800,
+        step: 50,
+      }),
+      size: select('Dropdown size (size)', sizes, INPUT_SIZE.MEDIUM),
+      readonly: boolean('Read only (readonly)', false),
+      type: textNullable('Type (type)', 'text'),
+      warn: boolean('Warn (warn)', false),
+      warnText: textNullable(
+        'Warn text (warn-text)',
+        'Warning message that is really long can wrap to more lines but should not be excessively long.'
+      ),
+      value: textNullable('Value of input (value)', ''),
+      onInput: action(`${prefix}-select-selected`),
+    }),
+  },
+};
+
+export default {
+  title: 'Components/TextInput',
```

Comment:
```suggestion
  title: 'Components/Text Input',
```
capital or lower case? might have some inconsistencies